geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ConnectCommand.java

```diff
@@ -168,26 +168,30 @@ public ResultModel connect(
       return result;
     }
 
-    // since 1.14, only allow gfsh to connect to cluster that's older than 1.10
     String remoteVersion = null;
-    String gfshVersion = gfsh.getVersion();
+    String remoteGeodeSerializationVersion = null;
     try {
       remoteVersion = invoker.getRemoteVersion();
-      int minorVersion = Integer.parseInt(versionComponent(remoteVersion, VERSION_MINOR));
-      if (versionComponent(remoteVersion, VERSION_MAJOR).equals("1") && minorVersion >= 10 ||
-          versionComponent(remoteVersion, VERSION_MAJOR).equals("9") && minorVersion >= 9) {
-        InfoResultModel versionInfo = result.addInfo("versionInfo");
-        versionInfo.addLine("You are connected to a cluster of version: " + remoteVersion);
-        return result;
+      try {
+        remoteGeodeSerializationVersion = invoker.getRemoteGeodeSerializationVersion();
```

Comment:
I think we should try to get the geodeSeriliazationVersion first since that's introduced later than the remoteVersion. If we can't get the remoteVersion, we definitely would fail at getting the remoteGeodeSerilizationVersion

Reply:
both are needed before calling `isCompatible`. so the only optimization is to skip the 2nd call to get remote serialization version if the first call to get product version failed (as this would indicate a pre-1.5 cluster or some network issue).  I can remove this optimization if you think it would make the code more readable.

Reply:
I can't tell what change you are asking for.  This PR already gets the remote version first, because there would be no reason to try to get remote serialization version if that fails...

Reply:
I think I misread your code.

Reply:
Can you put these two remote call in the same try block, this "try" inside "try" seems confusing:

```
try {
      // serialization version is  introduced later than remoteVersion, have to call them in the right order
      remoteVersion = invoker.getRemoteVersion();
      remoteSerializationVersion = invoker.getRemoteGeodeSerializationVersion();
    }
    catch (Exception ex) {
      // if unable to get the remote version, we are certainly talking to a pre-1.5 cluster
      // if unable to get the remote serilization version, we are certainly talking to a pre-1.12 cluster
      gfsh.logInfo("failed to get the the remote version.", ex);
    }
```

Reply:
combining like this would cause gfsh to print "failed to get the the remote version." when connecting to 1.10 or 1.11

Reply:
Oh, we use gfsh to log, maybe instead of calling `gfsh.logInfo` we should just call `gfsh.logToFile`

Reply:
factored out a method to make this more readable (and add the comment on ignored exception that Bill requested)